src/sagemaker/jumpstart/utils.py

```diff
@@ -136,3 +138,53 @@ def is_jumpstart_model_input(model_id: Optional[str], version: Optional[str]) ->
             )
         return True
     return False
+
+
+def is_jumpstart_model_uri(uri: Optional[str]) -> bool:
+    """Returns True if URI corresponds to a JumpStart-hosted model.
+
+    Args:
+        uri (Optional[str]): uri for inference/training job.
+    """
+
+    bucket = None
+    if urlparse(uri).scheme == "s3":
+        bucket, _ = parse_s3_url(uri)
+
+    return bucket in constants.JUMPSTART_BUCKET_NAME_SET
+
+
+def add_jumpstart_tags(
+    tags: Optional[List[Dict[str, str]]],
+    inference_model_uri: Optional[str],
+    inference_script_uri: Optional[str],
+) -> List[Dict[str, str]]:
+    """Adds tags for JumpStart models. Returns original tags for non-JumpStart
+    models.
+
+    Args:
+        tags (Optional[List[Dict[str,str]]): Current tags for JumpStart inference
+        or training job.
+        inference_model_uri (Optional[str]): S3 URI for inference model artifact.
+        inference_script_uri (Optional[str]): S3 URI for inference script tarball.
+    """
+
+    if is_jumpstart_model_uri(inference_model_uri):
+        if tags is None:
+            tags = []
+        tags.append(
+            {
+                constants.JumpStartTag.INFERENCE_MODEL_URI.value: inference_model_uri,
+            }
+        )
+
+    if is_jumpstart_model_uri(inference_script_uri):
+        if tags is None:
+            tags = []
+        tags.append(
+            {
+                constants.JumpStartTag.INFERENCE_SCRIPT_URI.value: inference_script_uri,
+            }
+        )
```

Comment:
I know this is an edge case, but could you check that these tags do not already exist, and not add them if that's the case? You would get a validation error if you tried to add two tags with the same key.

Reply:
Good idea